adapters/repos/db/vector/hnsw/search.go

```diff
@@ -806,10 +824,11 @@ func (h *hnsw) knnSearchByVector(ctx context.Context, searchVec []float32, k int
 		helpers.AnnotateSlowQueryLog(ctx, "knn_search_rescore_took", took)
 	}
 
-	for res.Len() > k {
-		res.Pop()
+	if !h.multivector.Load() {
```

Comment:
Why don't we limit our search to k elements when using multi-vector?

Reply:
Those are the vectors used to build the candidate set. Then are fetched the real documents, are computed the scores (maxSim) and finally are retrieved only the top-`k` docs.